flocker/node/agents/test/test_blockdevice.py

```diff
@@ -239,6 +239,55 @@ def test_overridden(self):
         self.assertIs(async_api, deployer.async_block_device_api)
 
 
+def assert_discovered_state(case,
+                            deployer, expected_hostname,
+                            expected_manifestations,
+                            expected_nonmanifest_datasets=None,
+                            expected_devices=pmap()):
+    """
+    Assert that the manifestations on the state object returned by
+    ``deployer.discover_state`` equals the given list of manifestations.
+
+    :param IDeployer deployer: The object to use to discover the state.
+    :param list expected_manifestations: The ``Manifestation``\ s expected
+        to be discovered.
+    :param dict expected_devices: The OS device files which are expected to
+        be discovered as allocated to volumes attached to the node.  See
+        ``NodeState.devices``.
+
+    :raise: A test failure exception if the manifestations are not what is
+            expected.
+    """
+    discovering = deployer.discover_state(
+        NodeState(hostname=expected_hostname)
```

Comment:
Merge forward and start using UUIDs for all new code creating `NodeState` and `Node`.